### PR TITLE
Clarify require and assert usage

### PR DESCRIPTION
**Pull request by maurelian**

### CODE REVIEW

**Comment by GNSPS** [comment]
The last typo you corrected has an extra `e` instead of having a fewer one, now. 😂

**Review by pirapira — APPROVED** [review]
Clearly an improvement.

File: docs/control-structures.rst [hunk 1]
```diff
@@ -381,13 +381,11 @@ Error handling: Assert, Require, Revert and Exceptions
...
-if the condition is not met. The difference between the two is that ``assert`` should only be used for internal errors
-and ``require`` should be used to check external conditions (invalid inputs or errors in external components).
-The idea behind that is that analysis tools can check your contract and try to come up with situations and
-series of function calls that will reach a failing assertion. If this is possible, this means there is a bug
-in your contract you should fix.
+if the condition is not met. The ``assert`` function should only be used to test for internal errors, and to check invariants.
+The ``require`` function should be used to ensure valid conditions, such as inputs, or contract state variables are met, or to validate return values from calls to external contracts. 
+If used properly, analysis tools can evaluate your contract to identify the conditions and function calls which will reach a failing ``assert``. Properly functioning code should never it is reach a failing assert statement, if this happens there is a bug in your contract which you should fix.
```
Comment (axic) [inline docs/control-structures.rst:386]: This part doesn't sound right "Properly functioning code should never it is reach a failing assert statement"

Comment (pirapira) [reply docs/control-structures.rst:386]: Sorry, see #2639 for my try to fix this.

Comment (maurelian) [reply docs/control-structures.rst:386]: Thank you @pirapira.

**Pull request merged**